webhooks-extension/pkg/endpoints/sink.go

```diff
@@ -242,19 +338,52 @@ func (r Resource) getPipelineImpl(name, namespace string) (v1alpha1.Pipeline, er
 	return *pipeline, nil
 }
 
+/* Get all tasks in a given namespace: the caller needs to handle any errors,
+an empty v1alpha1.Task{} is returned if no task is found */
+func (r Resource) getTaskImpl(name, namespace string) (v1alpha1.Task, error) {
+	logging.Log.Infof("In getTaskImpl, name %s, namespace %s.", name, namespace)
+
+	tasks := r.TektonClient.TektonV1alpha1().Tasks(namespace)
+	task, err := tasks.Get(name, metav1.GetOptions{})
+	if err != nil {
+		logging.Log.Errorf("error receiving the task called %s in namespace %s: %s.", name, namespace, err.Error())
+		return v1alpha1.Task{}, err
+	}
+	logging.Log.Info("Found the task definition OK.")
+	return *task, nil
+}
+
 /* Create a new PipelineResource: this should be of type git or image */
-func definePipelineResource(name, namespace string, params []v1alpha1.Param, resourceType v1alpha1.PipelineResourceType) *v1alpha1.PipelineResource {
+func definePipelineResource(name, namespace string, params []v1alpha1.Param, secrets []v1alpha1.SecretParam, resourceType v1alpha1.PipelineResourceType) *v1alpha1.PipelineResource {
 	pipelineResource := v1alpha1.PipelineResource{
 		ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: namespace},
 		Spec: v1alpha1.PipelineResourceSpec{
 			Type:   resourceType,
 			Params: params,
 		},
 	}
+	if secrets != nil {
+		pipelineResource.Spec.SecretParams = secrets
+	}
 	resourcePointer := &pipelineResource
 	return resourcePointer
 }
-
+/* Create a new PipelineResource: this should be of type git or image 
```

Comment:
Hmm can this now go or should we be using it instead of definePipelineResource?

Reply:
I remove this code.